airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/abstract_file_based_stream.py

```diff
@@ -101,7 +101,7 @@ def read_records(
         return self.read_records_from_slice(stream_slice)
 
     @abstractmethod
-    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[Mapping[str, Any]]:
+    def read_records_from_slice(self, stream_slice: StreamSlice) -> Union[Iterable[AirbyteMessage], Iterable[Mapping[str, Any]]]:
```

Comment:
are you changing the type because it was already wrong or because we're not returning AirbyteMessages in addition to mappings?